### PR TITLE
Adapt new or

### PR DESCRIPTION
**Pull request by u-fischer**
# Internal housekeeping

## Status of pull request

- Ready to merge

## Checklist of required changes before merge will be approved
- [x] Test file(s) added
- [x] Version and date string updated in changed source files
- [] Relevant `\changes` entries in source included
- [x] Relevant `changes.txt` updated
- [ ] `ltnewsX.tex` (and/or `latexchanges.tex`) updated

This moves some patches to the output routine from pdfmanagement-testphase to latex-lab as `new-or-1`. The other patches to the output routine are now called `new-or-2`, both are loaded with `new-or`. Some documentation has been corrected and updated.

### CODE REVIEW

File: required/latex-lab/changes.txt [hunk 1]
```diff
@@ -1,3 +1,11 @@
+2022-04-11 Ulrike Fischer <Ulrike.Fischer@latex-project.org>
+
+	* documentmeta-support.dtx:
+	updated documentation
+    * moved output patches from pdfresources to in a new latex-lab-new-or-1
```
Comment (FrankMittelbach) [inline required/latex-lab/changes.txt:5]: to new

File: required/latex-lab/documentmetadata-support.dtx [hunk 2]
```diff
@@ -64,13 +64,19 @@
...
-%    \cs{DocumentMetadata} is place in a separate bundle, so that it
+%    \cs{DocumentMetadata} is placed in a separate bundle, so that it
 %    is easier to update it without the need to build a full \LaTeX{}
 %    release. Over time the functionality will move fully into the
 %    kernel.
+%    
+%    \cs{DocumentMetadata} also loads and activates
+%    the new PDF management code from \pkg{pdfmanagement-testphase}.
+%    As this forces the loading of the \pkg{l3backend} files, a backend
+%    which can't be detected automatically like |dvipdfmx|,
```
Comment (FrankMittelbach) [inline required/latex-lab/documentmetadata-support.dtx:75]: which -> that

File: required/latex-lab/documentmetadata-support.dtx [hunk 3]
```diff
@@ -110,15 +118,34 @@
...
-%      but they will pass the information to hyperref and hyperxmp. The |a| variants
-%      do \emph{not} enforce (or even test) a tagged pdf yet. The |A-4| support is
-%      currently incomplete.
+%      Note that using these
+%      key doesn't mean that the document actually follows the standard. \LaTeX{}
+%      can neither ensure nor check all requirements of a standard, and not everything
+%      it can do theoretically has already been implemented.
+%      When setting an A-standard a color profile is included and the
+%      \texttt{/OutputIntent} is set and javascript action in hyperref are suppressed.
+%      The |u| variants do not force unicode,
```
Comment (FrankMittelbach) [inline required/latex-lab/documentmetadata-support.dtx:127]: enforce

File: required/latex-lab/documentmetadata-support.dtx [hunk 4]
```diff
@@ -110,15 +118,34 @@
...
+%      but they will pass the information to hyperref. The |a| variants
+%      do \emph{not} enforce (or even test) a tagged pdf yet.
+%      
+%      Starting with version 0.95s of \pkg{pdfmanagement-testphase}
+%      it is also possible to use the values
+%      |X-4|, |X-4p|, |X-5g|, |X-5n|, |X-5pg|, |X-6|, |X-6n|, |X-6p|, |UA-1| for
+%      a PDF/X and PDF/UA standard. These keys set \emph{only} the relevant
```
Comment (FrankMittelbach) [inline required/latex-lab/documentmetadata-support.dtx:134]: keys currently set

File: required/latex-lab/documentmetadata-support.dtx [hunk 5]
```diff
@@ -110,15 +118,34 @@
...
+%      
+%      If XMP-metadata are added (see the following key \texttt{xmp}
+%      the needed conformance marker for the standards are set.
+%      
 %      More information can be found in the documentation
-%      of \pkg{l3pdfmeta}.
+%      of \pkg{l3pdfmeta}.   
 %
+%    \item[\texttt{xmp}] A boolean, if set to false no XMP metadata are added to the PDF.
+%    The default is true. Details are described in the documentation of \pkg{l3pdfmeta}.
```
Comment (FrankMittelbach) [inline required/latex-lab/documentmetadata-support.dtx:147]: true even is not set or true if set without a value?

File: required/latex-lab/documentmetadata-support.dtx [hunk 6]
```diff
@@ -137,6 +164,16 @@
...
+%    of paragraphs, for the tagging of header and footer and 
+%    to allow the PDF management to insert code which avoids that
+%    links happening at page breaks spills into the header and footer. 
+%    This code is automatically loaded if the \texttt{testphase} values
+%    \texttt{phase-I}, \texttt{phase-II} or \texttt{new-or} are used.%   
+%    \item[\texttt{new-or}] This loads more changes to the output routine required for the 
+%    tagging. It is not compatible with every class! The code is also loaded by the 
+%    \texttt{phase-II} value.
 %    \end{description}
 %    The |testphase| key can only be used in the first \cs{DocumentMetadata}.
```
Comment (FrankMittelbach) [inline required/latex-lab/documentmetadata-support.dtx:178]: first call to

File: required/latex-lab/latex-lab-new-or-1.dtx [hunk 7]
```diff
@@ -0,0 +1,164 @@
+% \iffalse meta-comment
+%
+%% File: latex-lab-new-or-1.dtx (C) Copyright 2020-2022 Frank Mittelbach
```
Comment (FrankMittelbach) [inline required/latex-lab/latex-lab-new-or-1.dtx:3]: , Ulrike Fischer

File: required/latex-lab/latex-lab-new-or-1.dtx [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+%
+%
+% \title{Temporary output patches for tagging and better link support}
+% \author{Frank Mittelbach \and Ulrike Fischer}
+%
+% \maketitle
+%
+% \section{Introduction}
+%
+%    This package contains a few patches of internal \LaTeX{} needed
```
Comment (FrankMittelbach) [inline required/latex-lab/latex-lab-new-or-1.dtx:35]: This file    This code  (not package)

File: required/latex-lab/latex-lab-new-or-1.dtx [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+%
+%    \begin{macrocode}
+\ProvidesPackage{latex-lab-testphase-new-or-1}
+        [2022-11-94 v0.1a Temporary output patches for tagging and better link support]
+\RequirePackage{etoolbox}
+%    \end{macrocode}
+%
+%
+%  \subsection{Patching the OR}
+%  This add hooks around header and footer, in \cs{@makecol} and in multicol.
```
Comment (FrankMittelbach) [inline required/latex-lab/latex-lab-new-or-1.dtx:61]: This adds kernel hooks

File: required/latex-lab/latex-lab-new-or-1.dtx [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+%      update 255 up front to add missing tmb/tme if necessary
+%    \item
+%      update the \cs{footins} stream if that one got split
+%    \end{itemize}
+%
+%   If the new latex-lab file for the output routine is loaded it will do the right thing.
+%   In the other case we patch \cs{@makecol} manually.
+%    \begin{macrocode}
+\@ifpackageloaded{latex-lab-testphase-new-or}
+   {}
```
Comment (FrankMittelbach) [inline required/latex-lab/latex-lab-new-or-1.dtx:99]: is this still the right test?

File: required/latex-lab/latex-lab-new-or-2.dtx [hunk 11]
```diff
@@ -129,7 +129,7 @@
 % \subsection{File declaration}
 %    \begin{macrocode}
 \ProvidesFile{latex-lab-new-or.ltx}
```
Comment (FrankMittelbach) [inline required/latex-lab/latex-lab-new-or-2.dtx:131]: not the same as file name

File: required/latex-lab/testfiles-OR/footmisc-001.tlg [hunk 12]
```diff
@@ -60,4 +60,7 @@ Completed box being shipped out [1]
 ....\glue 0.0 plus 1.0fil
 ..\pdfrunninglinkon
 .\kern 0.0
+.\kern -633.0
+.\hbox(0.0+0.0)x0.0
+.\kern 633.0
```
Comment (FrankMittelbach) [inline required/latex-lab/testfiles-OR/footmisc-001.tlg:65]: where is that coming from? just curious

**Review by FrankMittelbach — APPROVED** [review]
not checked the code block though

File: required/latex-lab/testfiles-OR/footmisc-001.tlg (hunk 12 above)
Comment (u-fischer) [reply required/latex-lab/testfiles-OR/footmisc-001.tlg:65]: The xmp-code uses the shipout/lastpage hook (for dvi based engine it must be in a box). And as soon as this hook is not empty you get this box:
~~~~
\documentclass{article}
\AddToHook{shipout/lastpage}{\typeout{something}}
\begin{document}
blub\showoutput
\end{document}
~~~~

I will correct the rest.

File: required/latex-lab/latex-lab-new-or-1.dtx (hunk 10 above)
Comment (u-fischer) [reply required/latex-lab/latex-lab-new-or-1.dtx:99]: yes. It is not strictly needed, but if both new-or-1 and new-or-2 are loaded through new-or, then it is not needed to patch `\@makecol` as new-or-2 will overwrite `\@makecol` again later.

Comment (u-fischer) [reply required/latex-lab/latex-lab-new-or-1.dtx:99]: but you are right. It doesn't work anymore as there is no new-or.sty, only a ltx. So I remove that.

**Pull request merged**